hibernate-core/src/main/java/org/hibernate/proxy/pojo/bytebuddy/ByteBuddyProxyFactory.java

```diff
@@ -93,8 +92,11 @@ public HibernateProxy getProxy(
 			return proxy;
 		}
 		catch (Throwable t) {
-			LOG.error( LOG.bytecodeEnhancementFailed( entityName ), t );
-			throw new HibernateException( LOG.bytecodeEnhancementFailed( entityName ), t );
+			String logMessage = ( t.getCause() instanceof NoSuchMethodException ) ?
+					LOG.bytecodeEnhancementFailedBecauseOfDefaultConstructor( entityName ) :
+					LOG.bytecodeEnhancementFailed( entityName );
+			LOG.error( logMessage, t );
+			throw new HibernateException( logMessage, t );
```

Comment:
@gsmet I'll try to apply your suggestion.

Reply:
@gsmet If we skip the log as you suggest, then the log message will look like this:

> INFO DefaultLoadEventListener:132 - HHH000327: Error performing load command : org.hibernate.HibernateException: HHH000143: Bytecode enhancement failed because no public, protected or package-private default constructor was found for entity: org.hibernate.jpa.test.callbacks.PrivateConstructorTest$Parent. Private constructors don't work with runtime proxies!

I think it's better to have a separate error log entry as before as it makes it more clear what happened.

Reply:
Two things:
- as for the first error, I see how the way it is done would lead to this situation: we need to resolve the constructor in `postInstantiate()`. This way you will have the error once and for all when getting the constructor and you can use the pattern I mentioned.
- you can then revert the change you made to the error handling in `getProxy()` and keep it as it was.

As much as possible, we need to avoid the reflection calls at runtime.

Sorry for not seeing it right away.

Reply:
FWIW, I'm working on it.